src/Microsoft.AspNetCore.Mvc.TagHelpers/SelectTagHelper.cs

```diff
@@ -70,6 +72,12 @@ public override void Init(TagHelperContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (For == null && Items != null)
```

Comment:
Remove `&& Items != null`. At least one of the attributes _must_ be bound. And this will then match what is in the `Process()` method.